### PR TITLE
[Feature] add: the propertyOrder field to specify order in DataSource settings

### PR DESCRIPTION
**Pull request by rmakulov**
```json
            "properties": {
                "user": {
                    "type": "string"
                },
                "password": {
                    "type": "string"
                }
```

screen shot 2017-02-27 at 15 53 41

```json
            "properties": {
                "user": {
                    "type": "string",
                    "propertyOrder": 0
                },
                "password": {
                    "type": "string",
                    "propertyOrder": 1
                }
```

screen shot 2017-02-27 at 15 56 53

### CODE REVIEW

**Comment by arikfr** [comment]
Thanks! This is something that was on my mind for quite some time. But I wonder if we should use this approach and rely on the fact that OrderedDict gets serialized in order (and then the browser keeps this order)? This is how it works, but to the best of my knowledge this is not a guaranteed behavior. 

An alternative I was thinking of is a field that defines an order (something like `order: ["host", "username", "password", ...]`).

What do you think?

**Comment by rmakulov** [comment]
I also thought about that. I prefer the alternative you mentioned above.
Fields that are not specified are added at the end.

```json
            "order": ['host', 'port', 'user', 'password'],
            "required": ["dbname"],
            "secret": ["password"]

```

screen shot 2017-02-27 at 15 56 53

**Comment by arikfr** [comment]
But we're still depend on unguaranteed behavior of JSON. I was thinking of handling the order on the client side (using this extra property to sort the keys on the client side).

**Comment by rmakulov** [comment]
What about such implementation?

File: redash/query_runner/__init__.py [hunk 1]
```diff
@@ -43,6 +45,13 @@ class InterruptException(Exception):
     pass
 
 
+def index(l, x):  # Allow non-full key list to be used in sorting
+    try:
+        return l.index(x)
+    except ValueError:
+        return sys.maxint, x
```
Comment (arikfr) [inline redash/query_runner/__init__.py]: This is no longer needed, right?

Comment (rmakulov) [inline redash/query_runner/__init__.py]: Right :)

**Comment by arikfr** [comment]
Thanks!

Apologies for the long wait.

**Pull request merged**